core/src/test/java/io/grpc/internal/MetricRecorderImplTest.java

```diff
@@ -0,0 +1,195 @@
+  private static final DoubleCounterMetricInstrument DOUBLE_COUNTER_INSTRUMENT =
+      new DoubleCounterMetricInstrument(0, "counter0", DESCRIPTION, UNIT, REQUIRED_LABEL_KEYS,
+          OPTIONAL_LABEL_KEYS, ENABLED);
+  private static final LongCounterMetricInstrument LONG_COUNTER_INSTRUMENT =
+      new LongCounterMetricInstrument(1, "counter1", DESCRIPTION, UNIT, REQUIRED_LABEL_KEYS,
+          OPTIONAL_LABEL_KEYS, ENABLED);
+  private static final DoubleHistogramMetricInstrument DOUBLE_HISTOGRAM_INSTRUMENT =
+      new DoubleHistogramMetricInstrument(2, "histogram1", DESCRIPTION, UNIT,
+          Collections.emptyList(), REQUIRED_LABEL_KEYS, OPTIONAL_LABEL_KEYS, ENABLED);
+  private static final LongHistogramMetricInstrument LONG_HISTOGRAM_INSTRUMENT =
+      new LongHistogramMetricInstrument(3, "histogram2", DESCRIPTION, UNIT,
+          Collections.emptyList(), REQUIRED_LABEL_KEYS, OPTIONAL_LABEL_KEYS, ENABLED);
+  private final List<MetricInstrument> instruments = Arrays.asList(DOUBLE_COUNTER_INSTRUMENT,
+      LONG_COUNTER_INSTRUMENT, DOUBLE_HISTOGRAM_INSTRUMENT, LONG_HISTOGRAM_INSTRUMENT);
+  private final Set<String> metricNames = Stream.of("counter1", "counter2", "histogram1",
+          "histogram2")
+      .collect(Collectors.toCollection(HashSet::new));
+  private MetricSink mockSink = mock(MetricSink.class);
+  private List<MetricSink> sinks = Arrays.asList(mockSink, mockSink);
+  private MetricInstrumentRegistry registry;
+  private MetricRecorder recorder;
+  private ArgumentCaptor<DoubleCounterMetricInstrument> doubleCounterInstrumentCaptor =
+      ArgumentCaptor.forClass(DoubleCounterMetricInstrument.class);
+  private ArgumentCaptor<LongCounterMetricInstrument> longCounterInstrumentCaptor =
+      ArgumentCaptor.forClass(LongCounterMetricInstrument.class);
+  private ArgumentCaptor<DoubleHistogramMetricInstrument> doubleHistogramInstrumentCaptor =
+      ArgumentCaptor.forClass(DoubleHistogramMetricInstrument.class);
+  private ArgumentCaptor<LongHistogramMetricInstrument> longHistogramInstrumentCaptor =
+      ArgumentCaptor.forClass(LongHistogramMetricInstrument.class);
+
+
+  @Before
+  public void setUp() {
+    registry = MetricInstrumentRegistryAccessor.createMetricInstrumentRegistry(instruments,
+        metricNames);
+    recorder = new MetricRecorderImpl(sinks, registry);
+  }
+
+  @Test
+  public void recordCounter() {
+    when(mockSink.getMetricsMeasures()).thenReturn(
+        Arrays.asList(new Object(), new Object(), new Object(), new Object()));
+
+    recorder.recordDoubleCounter(DOUBLE_COUNTER_INSTRUMENT, 1.0, REQUIRED_LABEL_VALUES,
+        OPTIONAL_LABEL_VALUES);
+    verify(mockSink, times(2)).recordDoubleCounter(doubleCounterInstrumentCaptor.capture(), eq(1D),
+        argThat(l -> hasItems("VALUE1", "VALUE2").matches(l)),
+        argThat(l -> hasItems("OPTIONAL_VALUE_1").matches(l)));
+    assertThat(doubleCounterInstrumentCaptor.getValue()).isEqualTo(DOUBLE_COUNTER_INSTRUMENT);
```

Comment:
No need for the captor. Just pass `eq(DOUBLE_COUNTER_INSTRUMENT)` to the verify.